### DISCUSSION TITLE
ToBitmapSource and Microsoft.UI.Xaml.Media.ImageSource

### DISCUSSION THREAD

**Discussion by HEIC-to-JPEG-Dev** [dlemstra/Magick.NET#1411] [Help] [answered]
Currently, ToBitmapSource returns a System.Windows.Media.ImageSource

I currently convert this to the latest model (Microsoft.UI.Xaml.Media.ImageSource) by writing it to a MemoryStream and recreating a source from that.

Is there a better way, or a way for MagickImage to return the newer ImageSource ?

Thanks

**Comment by dlemstra** [comment]
I have special libraries for this as you have noticed. Maybe I could add an extra library for this if the api of the type allows that? But for me it's now a bit unclear which library contains that type.

**Comment by HEIC-to-JPEG-Dev** [comment]
WinUI (WASDK) is where it's used, also UWP apps using WinRT.

https://github.com/microsoft/WindowsAppSDK

**Comment by HEIC-to-JPEG-Dev** [comment, marked as answer]
I now use output to a memoryStream and create a BitmapImage from that - it seems the fastest way to convert the class. thanks
